okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java

```diff
@@ -118,11 +122,19 @@ public HttpLoggingInterceptor() {
   }
 
   public HttpLoggingInterceptor(Logger logger) {
+    this(logger, Collections.<String>emptyList());
+  }
+
+  public HttpLoggingInterceptor(Logger logger, List<String> headersToRedact) {
```

Comment:
Do you mind replacing this constructor with a method that adds a header to redact? I'm allergic to telescoping constructor parameters.

Reply:
I'm allergic to them either, but avoided such a method so that `headersToRedact` doesn't have to become synchronized. Do you prefer a synchronized set or do you think one is not necessary?

Reply:
We could have our cake and eat it too.

```
private volatile Set<String> headersToRedact = Collections.emptySet();

public void redactHeader(String name) {
  Set<String> newHeadersToRedact = new TreeSet<>(CASE_INSENSITIVE_ORDER);
  newHeadersToRedact.addAll(headersToRedact);
  newHeadersToRedact.add(name);
  this.headersToRedact = newHeadersToRedact;
}
```